components/uid.rst

```diff
@@ -0,0 +1,200 @@
+.. index::
+   single: UID
+   single: Components; UID
+
+The UID Component
+=================
+
+    The UID component provides utilities to work with `unique identifiers`_ (UIDs)
+    such as UUIDs and ULIDs.
+
+.. versionadded:: 5.1
+
+    The UID component was introduced in Symfony 5.1 as an
+    :doc:`experimental feature </contributing/code/experimental>`.
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/uid
+
+.. include:: /components/require_autoload.rst.inc
+
+UUIDs
+-----
+
+`UUIDs`_ (*universally unique identifiers*) are one of the most popular UIDs in
+the software industry. UUIDs are 128-bit numbers usually represented as five
+groups of hexadecimal characters: ``xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx``
+(the ``M`` digit is the UUID version and the ``N`` digit is the UUID variant).
+
+Generating UUIDs
+~~~~~~~~~~~~~~~~
+
+Use the named constructors of the ``Uuid`` class or any of the specific classes
+to create each type of UUID::
+
+    use Symfony\Component\Uid\Uuid;
+    use Symfony\Component\Uid\UuidV1;
+    use Symfony\Component\Uid\UuidV3;
+    use Symfony\Component\Uid\UuidV4;
+    use Symfony\Component\Uid\UuidV5;
+    use Symfony\Component\Uid\UuidV6;
+
+    // UUID type 1 generates the UUID using the MAC address of your device and a timestamp.
+    // Both are obtained automatically, so you don't have to pass any constructor argument.
+    $uuid = Uuid::v1();
+    $uuid = new UuidV1();
+
+    // UUID type 4 generates a random UUID, so you don't have to pass any constructor argument.
+    $uuid = Uuid::v4();
+    $uuid = new UuidV4();
+
+    // UUID type 3 and 5 generate a UUID hashing the given namespace and name. Type 3 uses
+    // MD5 hashes and Type 5 uses SHA-1. The namespace is another UUID (e.g. a Type 4 UUID)
+    // and the name is an arbitrary string (e.g. a product name; if it's unique).
+    $namespace = new UuidV4();
```

Comment:
then 4 again?

Reply:
Yes, UUIDv3 needs a namespace (which is another UUID). Easiest in this example is to just create a random one